pkg/asset/manifests/ibmcloud/cluster.go

```diff
@@ -69,9 +69,11 @@ func GenerateClusterAssets(installConfig *installconfig.InstallConfig, clusterID
 	// If no Control Plane subnets were provided in InstallConfig, we build a default set to cover all provided zones, or zones in the region.
 	if len(controlPlaneSubnets) == 0 {
 		var zones []string
-		// Use provided Control Plane zones, or default to all zones in the Region.
-		if len(installConfig.Config.ControlPlane.Platform.IBMCloud.Zones) != 0 {
+		// Use provided Control Plane zones, DefaultMachinePlatform zones, or default to all zones in the Region.
+		if installConfig.Config.ControlPlane.Platform.IBMCloud != nil && installConfig.Config.ControlPlane.Platform.IBMCloud.Zones != nil && len(installConfig.Config.ControlPlane.Platform.IBMCloud.Zones) != 0 { //nolint: gocritic //checking multiple fields are set and populated
 			zones = installConfig.Config.ControlPlane.Platform.IBMCloud.Zones
+		} else if platform.DefaultMachinePlatform != nil && platform.DefaultMachinePlatform.Zones != nil && len(platform.DefaultMachinePlatform.Zones) != 0 {
+			zones = platform.DefaultMachinePlatform.Zones
```

Comment:
```suggestion
		if ctrplane := installConfig.Config.ControlPlane; ctrplane.Platform.IBMCloud != nil && len(ctrplane.Platform.IBMCloud.Zones) > 0 {
			zones = installConfig.Config.ControlPlane.Platform.IBMCloud.Zones
		} else if platform.DefaultMachinePlatform != nil && len(platform.DefaultMachinePlatform.Zones) > 0 {
			zones = platform.DefaultMachinePlatform.Zones
```

I think the nil check here is only needed for the `Platform.IBMCloud` to fix the bug, right? The `len` func will return 0 length for `nil` slice.

Reply:
That was my assumption as well, but per this comment, it seems like both `nil` checks are required
https://github.com/openshift/installer/pull/9415#pullrequestreview-2628809355

I can make the assignment change however.

Reply:
updated

Reply:
need add installConfig.Config.ControlPlane.Platform.IBMCloud.Zones!=nil

Reply:
Thanks @MayXuQQ! That still surprises me (i.e. purely based on golang `len` behavior) but I suppose it is, for some reasons, needed here.